src/reducers/bucketlistReducer.js

```diff
@@ -0,0 +1,104 @@
+import {
+  ADD_BUCKETLIST_SUCCESS,
+  ADD_BUCKETLIST_FAILURE,
+  ADD_BUCKETLIST_REQUEST,
+  GET_BUCKETLISTS_REQUEST,
+  GET_BUCKETLISTS_SUCCESS,
+  GET_BUCKETLISTS_FAILURE,
+  UPDATE_BUCKETLIST_REQUEST,
+  UPDATE_BUCKETLIST_SUCCESS,
+  UPDATE_BUCKETLIST_FAILURE,
+  DELETE_BUCKETLIST_SUCCESS,
+  DELETE_BUCKETLIST_FAILURE,
+  DELETE_BUCKETLIST_REQUEST,
+  RESET_REQUEST_STATE
+} from "../actions/constants";
+
+const initialState = {
+  request: {
+    "status": "",
+    "message": ""
+  }
+}
+
+export default function bucketlists(state : Object = initialState, action) {
+  switch (action.type) {
+    
+    case GET_BUCKETLISTS_REQUEST:
+    case UPDATE_BUCKETLIST_REQUEST:
+    case DELETE_BUCKETLIST_REQUEST:
+    case ADD_BUCKETLIST_REQUEST:
+      return state;
+
+    case RESET_REQUEST_STATE:
+      return Object.assign([], state, {
```

Comment:
I think `Array#concat` is better suited to immutably add an object to an array of objects instead of `Object.assign`.

Reply:
I'll change this to concat instead